src/models/thread.ts

```diff
@@ -358,6 +358,55 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         this.pendingReplyCount = pendingEvents.length;
     }
 
+    /**
+     * Resets the live timeline if we're desynced.
+     *
+     * During connection failures, sometimes not all sync state for a thread is saved.
+     * After a reconnect, the thread will be missing some messages.
+     * We try to prevent that by detecting such cases and resetting the live timeline accordingly.
+     * @private
+     */
+    private async tryResetLiveTimeline(): Promise<void> {
+        // if we're still syncing, let the sync complete first
+        if (!this.client.isInitialSyncComplete()) {
+            return;
+        }
+
+        const lastKnownEvent = this.events[this.events.length - 1];
```

Comment:
This won't work, because the message in the bundled relationship of the root event is only going to be the last event with a `thread` relation type.
However the last known event in the timeline could be a reaction that is a second degree relation to a thread event